replaces repl iframe with new sandbox (#5)
learn-co-curriculum/phase-0-pac-1-logical-operators: README.md

The logical NOT operator (`!`), also called the _bang operator_, operates on an
expression, returning the opposite of the expression's truthiness. If `x`
resolves to a truthy value, `!x` returns `false`. If `x` is falsey, `!x` returns
`true`. Let's add the following code into our REPL and see what the values are.

```js
const truthyValue = 'This value is truthy.';
const falseyValue = 0;

!truthyValue;
```

<iframe height="400px" width="100%" src="https://replit.com/@lizbur10/Sandbox?lite=1&outputonly=1" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>

If you click the Run button you'll see the bang operator in action, returning
the reverse of `truthyValue`'s truthiness. Then modify the code on the last line
to verify that it works for `falseyValue` as well.

> 

Reviewer: We should remove the reference to clicking the Run button - perhaps "When you hit enter" instead. I'd also remove the instruction to modify the code since there isn't really anything modifiable. I think we can just say "Verify that it works for ..."

The logical NOT operator (`!`), also called the _bang operator_, operates on an
expression, returning the opposite of the expression's truthiness. If `x`
resolves to a truthy value, `!x` returns `false`. If `x` is falsey, `!x` returns
`true`. Let's add the following code into our REPL and see what the values are.

```js
const truthyValue = 'This value is truthy.';
const falseyValue = 0;

!truthyValue;
```

<iframe height="400px" width="100%" src="https://replit.com/@lizbur10/Sandbox?lite=1&outputonly=1" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>

Verify the above code works by running it in the REPL console (remember to hit
`enter`!) Play around with it and you'll see the bang operator in action,
returning the reverse of `truthyValue`'s truthiness. Then modify the code on the

> last line to verify that it works for `falseyValue` as well.

Reviewer: Let's say something like "Be sure to verify that it works for `falseyValue` as well" instead - the "modify the code" phrase seems a little off without a code editor.

Author: ```suggestion
Verify the above code works by running it in the REPL console (remember to hit
`enter`!) Play around with it and you'll see the bang operator in action,
returning the reverse of `truthyValue`'s truthiness. Be sure to verify that it works for `falseyValue` as well.
```

The logical NOT operator (`!`), also called the _bang operator_, operates on an
expression, returning the opposite of the expression's truthiness. If `x`
resolves to a truthy value, `!x` returns `false`. If `x` is falsey, `!x` returns
`true`. Let's add the following code into our REPL and see what the values are.

```js
const truthyValue = 'This value is truthy.';
const falseyValue = 0;

!truthyValue;
```

<iframe height="400px" width="100%" src="https://replit.com/@lizbur10/Sandbox?lite=1&outputonly=1" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>

Verify the above code works by running it in the REPL console (remember to hit
`enter`!) Play around with it and you'll see the bang operator in action,
returning the reverse of `truthyValue`'s truthiness. Then modify the code on the
Verify the above code works by running it in the REPL console (remember to hit

> `enter`!) Play around with it and you'll see the bang operator in action,

Reviewer: Lines 40-42 need to be removed

<iframe height="400px" width="100%" src="https://replit.com/@lizbur10/Sandbox?lite=1&outputonly=1" scrolling="no" frameborder="no" allowtransparency="true" allowfullscreen="true" sandbox="allow-forms allow-pointer-lock allow-popups allow-same-origin allow-scripts allow-modals"></iframe>

> Verify the above code works by running it in the REPL console (remember to hit

Author: ```suggestion
Verify the above code works by running it in the REPL console (remember to hit
`enter`!) Play around with it and you'll see the bang operator in action,
returning the reverse of `truthyValue`'s truthiness. Be sure to verify that it works with the
`falseyValue` as well.
```

constructed Boolean.

As a shorter way to convert any value into a Boolean, we can use two NOT

> operators. Let's change the code on the last line to `!!truthyValue`.

Reviewer: Here again there's nothing to change really - they just need to run the code

## Practicing What We've Learned

The code below contains a number of expressions that use the logical operators
we've learned about in this lesson:

```js
//What should each of the following expressions return? Once you have an answer, copy the expression in question into the console and hit enter to see if you're right!

// Practice with AND
0 && false;

// 0 && true;

// true && NaN;

// true && !1;

// !0 && "This is a string";

// !0 && "";

// !0 && !!"";

// Practice with OR

// 0 || false;

// true || false;

// true || 1;

// !true || !false

// !1 || !0
```

> 

Reviewer: There's no real reason to comment out the code lines here now that we're not using the repl. I think it'll be easier for students to read and understand if just the actual comments are commented out. You'll need to update the instructions to match.

We also need to edit the "A Note About Comments" section. I suggest changing the first sentence to "Note that some of the lines in the code above currently...", adding a sentence to the end of the first paragraph: "You'll learn more about this later.", and removing the second paragraph. 

Unfortunately, without the repl, there's no way to illustrate the use of comments to keep code from executing. We'll need to find a place to cover that later in the curriculum.

we've learned about in this lesson:

```js
//What should each of the following expressions return? Once you have an answer, copy the expression in question into the console and hit enter to see if you're right!

// Practice with AND
0 && false;

0 && true;

true && NaN;

true && !1;

!0 && "This is a string";

!0 && "";

!0 && !!"";

// Practice with OR
0 || false;

true || false;

true || 1;

!true || !false

!1 || !0
```

Before we get to that, however...

### A Note About Comments in Code

Note that some of the lines in the code above currently have the characters `//`
at the front. The `//` is used in JavaScript to 'comment out' lines of code. Any
line in your code that has `//` at the front will _not be interpreted_ by the
JavaScript engine. This enables us to do two things: 1) include text (i.e.,
messages meant for humans like we have above) in our code files without
JavaScript throwing an error, and 2) _temporarily_ keep lines of code from being
interpreted; this second use can be especially handy when debugging code. You'll
learn more about this later.

### Instructions

Okay, let's get some practice with logical operators. Take a look at the 4th

> line in the code above (the only un-commented out line). What do you think the

Reviewer: Let's change the second sentence to: "Take a look at the first un-commented out line in the snippet above."

Author: ```suggestion 
Okay, let's get some practice with logical operators. Take a look at the first un-commented out line
line in the snippet above. What do you think the
```

expression will return? Think it through and come up with an answer, then copy &
paste it into your console and hit enter to check whether you're right. When
you're done with the first expression, work your way down, un-commenting out

> each expression in turn. Be sure you think each example through and figure out

Reviewer: We need to remove the "un-commenting out..." part. Maybe "work your way down through the remaining expressions" instead.

Author: ```suggestion
expression will return? Think it through and come up with an answer, then copy &
paste it into your console and hit enter to check whether you're right. When
you're done with the first expression, work your way down through the remaining 
expression in turn. Be sure you think each example through and figure out
```

!0 && "This is a string";

!0 && "";

!0 && !!"";

// Practice with OR
0 || false;

true || false;

true || 1;

!true || !false

!1 || !0
```

Before we get to that, however...

### A Note About Comments in Code

Note that some of the lines in the code above currently have the characters `//`
at the front. The `//` is used in JavaScript to 'comment out' lines of code. Any
line in your code that has `//` at the front will _not be interpreted_ by the
JavaScript engine. This enables us to do two things: 1) include text (i.e.,
messages meant for humans like we have above) in our code files without
JavaScript throwing an error, and 2) _temporarily_ keep lines of code from being
interpreted; this second use can be especially handy when debugging code. You'll
learn more about this later.

### Instructions

Okay, let's get some practice with logical operators. Take a look at the first
un-commented out line in the snippet above. What do you think the
expression will return? Think it through and come up with an answer, then copy &
paste it into your console and hit enter to check whether you're right. When
you're done with the first expression, work your way down, un-commenting out

> expression will return? Think it through and come up with an answer, then copy &

Reviewer: Lines 241-243 need to be deleted

paste it into your console and hit enter to check whether you're right. When
you're done with the first expression, work your way down through the remaining 

> expression in turn. Be sure you think each example through and figure out

Reviewer: Should be "expressions"

The code below contains a number of expressions that use the logical operators
we've learned about in this lesson:

```js
//What should each of the following expressions return? Once you have an answer, copy the expression in question into the console and hit enter to see if you're right!

// Practice with AND
0 && false;

0 && true;

true && NaN;

true && !1;

!0 && "This is a string";

!0 && "";

!0 && !!"";

// Practice with OR
0 || false;

true || false;

true || 1;

!true || !false

!1 || !0
```

Before we get to that, however...

### A Note About Comments in Code

Note that some of the lines in the code above currently have the characters `//`
at the front. The `//` is used in JavaScript to 'comment out' lines of code. Any
line in your code that has `//` at the front will _not be interpreted_ by the
JavaScript engine. This enables us to do two things: 1) include text (i.e.,
messages meant for humans like we have above) in our code files without
JavaScript throwing an error, and 2) _temporarily_ keep lines of code from being
interpreted; this second use can be especially handy when debugging code. You'll
learn more about this later.

### Instructions

> Okay, let's get some practice with logical operators. Take a look at the 4th

Author: ```suggestion
```
